### DISCUSSION TITLE
Converting TIFF files to TIFF is quite slow

### DISCUSSION THREAD

**Discussion by LyAnhVu** [ImageMagick/ImageMagick#7450] [General]
ImageMagick version
7.1.1-27

Operating system, version and so on
Windows 11

Description
When I use ImageMagick to convert tiff to tiff attached file, the processing time is quite long.
Please help me find out the root cause and solution to fix it. Thank you.

Command line
`magick.exe convert -enhance -monochrome -compress Group4 -quiet -define tiff:ignore-tags=32934 -strip -define tiff:preserve-compression=true <input file> <output file>`
[sample.zip](https://github.com/user-attachments/files/16063870/sample.zip)

**Comment by exekutive** [comment]
I get errors when I try

```
magick sample.tif -enhance -monochrome -compress Group4 -quiet \
 -define tiff:ignore-tags=32934 -strip \
 -define tiff:preserve-compression=true output.tif
magick: ASCII value for tag "Make" contains null byte in value; value incorrectly truncated during reading due to implementation limitations. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/945.
magick: ASCII value for tag "Model" contains null byte in value; value incorrectly truncated during reading due to implementation limitations. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/945.
```

**Reply by LyAnhVu** [reply]
Help me try again:
magick -enhance -monochrome -compress Group4 -quiet \
 -define tiff:ignore-tags=32934 -strip \
 -define tiff:preserve-compression=true tiff:sample.tif tiff:output.tif

I use libtiff version 4.6.0.

**Comment by exekutive** [comment]
the `-enhance -monochrome` operations seems to be what takes most of the time. If I take them out the process is instant. This is a 30 megapixel image after all.

**Reply by LyAnhVu** [reply]
I know that but I still want to keep those 2 options. `-enhance -monochrome`

**Reply by exekutive** [reply]
maybe try some different algorithms instead, like  `-despeckle` or `-mean-shift`

**Comment by snibgo** [comment]
> I get errors when I try
magick: ASCII value for tag "Make" contains null byte in value; value incorrectly truncated during reading due to implementation limitations. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/945.
magick: ASCII value for tag "Model" contains null byte in value; value incorrectly truncated during reading due to implementation limitations. `TIFFFetchNormalTag' @ warning/tiff.c/TIFFWarnings/945.

Those are warnings, not errors.

I suggest removing `convert` from your command, and rearranging options so they are in the order they should be processed: read the input, process it, write it. This will make no difference to the performance, but will help humans understand what the command does.
```
magick -quiet -define tiff:ignore-tags=32934 sample.tif -strip -enhance -monochrome -define tiff:preserve-compression=true out.tif
```
Removing the options `-enhance -monochrome` greatly increases the performance, and makes no difference to the output. You say "I still want to keep those 2 options." Why? I suggest you find input images where this does make a difference. There might be faster operations that do what you need.

If your build of IM is Q16 with HDRI, then a build that is Q8 without HDRI will be faster. It will also be less accurate, but probably makes no practical difference in this example.
